src/dataSources/cloudFirestore/asset.js

```diff
@@ -0,0 +1,228 @@
+import debug from 'debug';
+import { utility } from '@thatconference/api';
+import * as Sentry from '@sentry/node';
+
+const { dateForge } = utility.firestoreDateForge;
+const dlog = debug('that:api:assets:datasources:firebase');
+const collectionName = 'assets';
+const assignmentColName = 'assetAssignments';
+
+function scrubAsset({ asset, isNew, user }) {
+  const dirtyAsset = asset;
+  const now = new Date();
+  if (isNew) {
+    dirtyAsset.createdAt = now;
+    dirtyAsset.createdBy = user.sub;
+    if (!dirtyAsset.links) dirtyAsset.links = [];
+  }
+  dirtyAsset.lastUpdatedAt = now;
+  dirtyAsset.lastUpdatedBy = user.sub;
+
+  if (dirtyAsset.links)
+    dirtyAsset.links = dirtyAsset.links.map(l => ({
+      linkType: l.linkType,
+      url: l.url.toString(),
+    }));
+
+  return dirtyAsset;
+}
+
+// TODO move fixDates to that-api once Asset is more finalized
+function fixDates({ asset }) {
+  const assetOut = asset;
+  if (asset.createdAt) assetOut.createdAt = dateForge(asset.createdAt);
+  if (asset.lastUpdatedAt)
+    assetOut.lastUpdatedAt = dateForge(asset.lastUpdatedAt);
+  if (asset.startTime) assetOut.startTime = dateForge(asset.startTime);
+  if (asset.stopTime) assetOut.stopTime = dateForge(asset.stopTime);
+
+  return assetOut;
+}
+
+function assets(dbInstance) {
+  dlog('sessions data source created');
+  const assetCollection = dbInstance.collection(collectionName);
+  const assignmentCollection = dbInstance.collection(assignmentColName);
+
+  async function get(assetId) {
+    dlog('get asset %s', assetId);
+    const docSnap = await assetCollection.doc(assetId).get();
+    let result = null;
+    if (docSnap.exists) {
+      result = {
+        id: docSnap.id,
+        ...docSnap.data(),
+      };
+      result = fixDates({ asset: result });
+    }
+
+    return result;
+  }
+
+  function getBatch(ids) {
+    dlog('getBatch for %d asset ids', ids.length);
+    if (!Array.isArray(ids))
+      throw new Error('getBatch must receive an array of ids');
+
+    return Promise.all(ids.map(id => get(id)));
+  }
+
+  async function setAssignmentsBatch({
+    firestoreBatch,
+    user,
+    assignments,
+    assetId,
+  }) {
+    dlog('set assignments batch %o', assignments);
+    // TODO: verify that user may assign asset to given entity
```

Comment:
we should log a github task for this and below so we don't loose track of it.

Reply:
done. #98